src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.cpp

```diff
@@ -0,0 +1,167 @@
+  // std::cout << "Supported sub-commands are:\n";
+  // std::cout << "\tflash - Update SC firmware or shell on the device\n";
+  // std::cout << "\thelp - Print out help message for a sub-command\n";
+  // std::cout << "\tscan - List all detected mgmt PCIE functions\n";
+  // std::cout << "\tversion - Print out xrt build version\n";
+  // std::cout << "Experts only:\n";
+  // std::cout << "\tclock - Change various clock frequency on the device\n";
+  // std::cout << "\tconfig - Parse or update daemon/device configuration\n";
+  // std::cout << "\tnifd - Access the NIFD debug IP to readback frames and offsets\n";
+  // std::cout << "\tpartition - Show and download partition onto the device\n";
+  // std::cout << "\treset - Perform various flavors of reset on the device\n";
+  // std::cout << "Run xbmgmt --help <subcommand> for detailed help of each subcommand\n";
+
+}
+//to-do: check administrator
+
+// ------ Program entry point -------------------------------------------------
+ReturnCodes main_(int argc, char** argv) {
+
+  // Global options
+  bool bVerbose = false;
+  bool bTrace = false;
+  bool bHelp = false;
+  bool bOverride = false;
+
+  // Build our global options
+  po::options_description globalOptions("Global options");
+  globalOptions.add_options()
+    ("help", boost::program_options::bool_switch(&bHelp), "Help to use this program")
+    ("verbose", boost::program_options::bool_switch(&bVerbose), "Turn on verbosity")
+    ("trace", boost::program_options::bool_switch(&bTrace), "Enables code flow tracing")
+    ("override", boost::program_options::bool_switch(&bOverride), "Bypasses the development check for this executable")
+    ("command", po::value<std::string>(), "command to execute")
+    ("subArguments", po::value<std::vector<std::string> >(), "Arguments for command")
+  ;
+
+  // Create a sub-option command and arguments
+  po::positional_options_description positionalCommand;
+  positionalCommand.
+    add("command", 1 /* max_count */).
+    add("subArguments", 1 /* Unlimited max_count */);
+
+  // Parse the command line
+  po::parsed_options parsed = po::command_line_parser(argc, argv).
+    options(globalOptions).         // Global options
+    positional(positionalCommand).  // Our commands
+    allow_unregistered().           // Allow for unregistered options (needed for sub obtions)
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsed, vm);          // Can throw
+    po::notify(vm);                 // Can throw
+  } catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+    std::cerr << globalOptions << std::endl;
+    return RC_ERROR_IN_COMMAND_LINE;
+  }
+
+  if (bOverride == false) {
+    std::cout << std::endl;
+    std::cout << "===========================================================" << std::endl;
+    std::cout << "xbutil2 is currently under development and currently is    " << std::endl;

```

Comment:
Looks like copied code.  Can we share?  At least "xbutil2" should be xbmgmt2

Reply:
Hmm good point. Everything is the same except xbutil2->xbmgmt2. I'll refactor the code in the next PR